src/engine/blocks.js

```diff
@@ -589,6 +597,9 @@ class Blocks {
             } else {
                 // Changing the value in a dropdown
                 block.fields[args.name].value = args.value;
+                if (block.mutation && block.mutation.blockInfo) {
+                    block.mutation.blockInfo.arguments[args.name].selectedValue = args.value;
```

Comment:
I update the actual selected value in blockInfo mutation here so that the block still has the correct variable name selected after the following steps: 

1) Create new variable foo
2) Drag out `set [my variable] to (0)` block
3) Change the dropdown menu selection in that block to `foo`
4) Switch to the costumes tab and switch back

Without this change, switching back to the code tab would change the selected variable option in that block back to `my variable`. See discussion below for more info about why this is.